### PR TITLE
Kernel: General Interrupt Handling code cleanup

### PR DESCRIPTION
**Pull request by IdanHo**

### CODE REVIEW

File: Kernel/Arch/x86/common/InterruptManagement.cpp [hunk 1]
```diff
@@ -143,15 +143,7 @@ UNMAP_AFTER_INIT void InterruptManagement::switch_to_pic_mode()
     m_interrupt_controllers.append(adopt_lock_ref(*new PIC()));
     SpuriousInterruptHandler::initialize(7);
     SpuriousInterruptHandler::initialize(15);
-    for (auto& irq_controller : m_interrupt_controllers) {
-        VERIFY(irq_controller);
-        if (irq_controller->type() == IRQControllerType::i82093AA) {
-            irq_controller->hard_disable();
```
Comment (supercomputer7) [inline Kernel/Arch/x86/common/InterruptManagement.cpp:152]: Don't we need to hard disable the IOAPIC in case of switching to use the PIC?

**Review by linusg — COMMENTED** [review]
`Kernel: Simplify InterruptManagement::switch_to_pic_mode()` is missing some context IMO - why are the loop and hard_reset call suddenly gone?

File: Kernel/Arch/x86/common/InterruptManagement.cpp (hunk 1 above)
Comment (IdanHo) [reply Kernel/Arch/x86/common/InterruptManagement.cpp:152]: No, if we got to this function we don't have an IOAPIC in the `m_interrupt_controllers` list, so this should never happen.

**Comment by IdanHo** [comment]
> `Kernel: Simplify InterruptManagement::switch_to_pic_mode()` is missing some context IMO - why are the loop and hard_disable call suddenly gone?

See my answer to Liav above, I believe it's dead code.

File: Kernel/Arch/x86/common/InterruptManagement.cpp (hunk 1 above)
Comment (linusg) [reply Kernel/Arch/x86/common/InterruptManagement.cpp:152]: Ah, makes sense. Mind adding a sentence about that to the commit message?

Comment (IdanHo) [reply Kernel/Arch/x86/common/InterruptManagement.cpp:152]: Sure

Comment (supercomputer7) [reply Kernel/Arch/x86/common/InterruptManagement.cpp:152]: Please add `VERIFY(m_interrupt_controllers.size() == 1)` to ensure this stays true.

Comment (IdanHo) [reply Kernel/Arch/x86/common/InterruptManagement.cpp:152]: Done & done

**Review by supercomputer7 — APPROVED** [review]
Looks good to me, thanks for cleaning up this old code :)

**Review by linusg — APPROVED** [review]

**Pull request merged**